common/lib/xmodule/xmodule/library_content_module.py

```diff
@@ -219,25 +219,56 @@ def selected_children(self):
         if hasattr(self, "_selected_set"):
             # Already done:
             return self._selected_set  # pylint: disable=access-member-before-definition
+
+        lib_tools = self.runtime.service(self, 'library_tools')
+        format_block_keys = lambda block_keys: lib_tools.create_block_analytics_summary(self.location.course_key, block_keys)
+
         # Determine which of our children we will show:
         selected = set(tuple(k) for k in self.selected)  # set of (block_type, block_id) tuples
         valid_block_keys = set([(c.block_type, c.block_id) for c in self.children])  # pylint: disable=no-member
         # Remove any selected blocks that are no longer valid:
-        selected -= (selected - valid_block_keys)
+        invalid_block_keys = (selected - valid_block_keys)
+        if invalid_block_keys:
+            selected -= invalid_block_keys
+            # Publish an event for analytics purposes:
+            self.runtime.publish(self, "edx.librarycontentblock.content.removed", {
+                "location": unicode(self.location),
+                "removed": format_block_keys(invalid_block_keys),
+                "reason": "invalid",  # Deleted from library or library being used has changed
```

Comment:
Yes deletion is a change. But it's also possible that the block can be configured to use library A [blocks a,b,c] and then changed to use library B [blocks d,e,f], causing all of the blocks in the course to need to be replaced. But that's not what I would call a "deletion".

Also yes, course author changes can result in the subset of questions used changing from [a,b,c] to [a,c,d] due to different filters being specified.